protocol/otlp/metrics.go

```diff
@@ -0,0 +1,358 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This code is copied and modified directly from the OTEL Collector:
+// https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1d6309bb62264cc7e2dda076ed95385b1ddef28a/pkg/translator/signalfx/from_metrics.go
+
+package otlp
+
+import (
+	"encoding/json"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	metricsservicev1 "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	commonv1 "go.opentelemetry.io/proto/otlp/common/v1"
+	metricsv1 "go.opentelemetry.io/proto/otlp/metrics/v1"
+)
+
+var (
+	// Some standard dimension keys.
+	// upper bound dimension key for histogram buckets.
+	upperBoundDimensionKey = "upper_bound"
+
+	// infinity bound dimension value is used on all histograms.
+	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
+)
+
+// FromOTLPMetricRequest converts the ResourceMetrics in an incoming request to SignalFx datapoints
+func FromOTLPMetricRequest(md *metricsservicev1.ExportMetricsServiceRequest) []*datapoint.Datapoint {
+	return FromOTLPResourceMetrics(md.GetResourceMetrics())
+}
+
+// FromMetrics converts OTLP ResourceMetrics to SignalFx datapoints.
+func FromOTLPResourceMetrics(rms []*metricsv1.ResourceMetrics) []*datapoint.Datapoint {
+	var sfxDps []*datapoint.Datapoint
+
+	for _, rm := range rms {
+		for _, ilm := range rm.GetInstrumentationLibraryMetrics() {
```

Comment:
do we really care about this metrics ?

Reply:
how is the conversion happening please share document

Reply:
so basically we convert it our model right ?

Reply:
yes, exactly.